### ISSUE TITLE
Reservation bug when the reservation is too long

### ISSUE THREAD

**Issue by lolokai** [glpi-project/glpi#3993]
* Steps to reproduce (which actions have you made) :
Reserve a computer for about 2 years
Try to modify or delete the reservation

* Expected result :
You whould be able to modify the date

* Actual result :
Your browser becomes unavailable due to an ajax overload

* URL of the page :
https://GLPI/front/reservationitem.php

* Screenshot of the problem (if pertinent) :

* Your GLPI setup (you can find it in Setup > General menu, System tab) :
GLPI 9.2.3

**Comment by cedric-anne** [comment]
The problem is the update form displays all possible duration values between now and the end date, using the `time_step` configuration (5 minutes by default). For a 2 years duration, there is about 210k values.

The solution could be to display sames values as in creation form, preceded by the current value, but changing this will require to adapt all GLPI duration dropdowns. It could be done by commenting this line : https://github.com/glpi-project/glpi/blob/9.2/bugfixes/inc/dropdown.class.php#L1605

**Comment by trasher** [comment]
An alternative would be to replace the duration dropdown with a end resevation datetime. This implies to change the database structure and to convert all current durations into end datetime.

**Comment by trasher** [comment]
Or maybe use the select2 pager to limit what is currently loaded. We already use that to limit max values retrieved, don't know if we can also work with min ones.

**Comment by Phorms** [comment]
Hi,

we just delete the following line https://github.com/glpi-project/glpi/blob/9.4/bugfixes/inc/reservation.class.php#L712

That will skip the generation of the long list, but the dropdown needs a workaround. The end date shows only "Specify an end date", but no date picker. Select any other value from the dropdown and then select "Specify an end date" again. Now the date picker will show up.

That's how we use reservations, because we have reservations of several years and the list generation makes it unusable.

**Comment by flonou** [comment]
Hi, this is marked as completed but on 10.0.3 I still see this issue where a long reservation basically hangs the current web browser tab to compute all intervals.
From what I've checked on the 10.0 bugfix and main branches, this issue is not resolved. 
I would argue that using an end date to make/edit reservation is better when dealing with medium (let's say 1 day) to long reservations (but maybe most of use cases prefer intervals ?) . For now I just added 
`$default_delay = 0;`
to https://github.com/glpi-project/glpi/blob/b03fa609f719445026f2489731ef1bf7bab3170a/inc/reservation.class.php#L708 as we are mostly dealing with day/week/months/year long reservations.
